keep/api/routes/providers.py

```diff
@@ -621,36 +622,53 @@ def install_provider_webhook(
     secret_manager = SecretManagerFactory.get_secret_manager(context_manager)
     provider_secret_name = f"{tenant_id}_{provider_type}_{provider_id}"
     provider_config = secret_manager.read_secret(provider_secret_name, is_json=True)
+    provider_class = ProvidersFactory.get_provider_class(provider_type)
     provider = ProvidersFactory.get_provider(
         context_manager, provider_id, provider_type, provider_config
     )
     api_url = config("KEEP_API_URL")
     keep_webhook_api_url = (
         f"{api_url}/alerts/event/{provider_type}?provider_id={provider_id}"
     )
+    keep_webhook_incidents_api_url = (
+        f"{api_url}/incidents/event/{provider_type}?provider_id={provider_id}"
+    )
     webhook_api_key = get_or_create_api_key(
         session=session,
         tenant_id=tenant_id,
         created_by="system",
         unique_api_key_id="webhook",
         system_description="Webhooks API key",
     )
-
-    try:
-        extra_config = provider.setup_webhook(
-            tenant_id, keep_webhook_api_url, webhook_api_key, True
-        )
-        if extra_config:
-            provider_config["authentication"].update(extra_config)
-            secret_manager.write_secret(
-                secret_name=provider_secret_name,
-                secret_value=json.dumps(provider_config),
+    if issubclass(provider_class, BaseProvider) and provider_class.__dict__.get("setup_webhook") is not None:
+        try:
+            extra_config = provider.setup_webhook(
+                tenant_id, keep_webhook_api_url, webhook_api_key, True
             )
```

Comment:
or something to simplify all
```suggestion
            extra_config = provider.setup_incident_webhook(tenant_id, keep_webhook_incidents_api_url, webhook_api_key, True) if issubclass(provider_class, BaseIncidentProvider) else provider.setup_webhook(
                tenant_id, keep_webhook_api_url, webhook_api_key, True
            )
```